docker/scripts/server_init.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+# run migrations
+yarn migrate
+
+# TODO: investigate and resolve issue with local permission setup and then remove the following
+psql postgres://postgres:password@localhost:5432/server -c 'GRANT ALL PRIVILEGES ON TABLE credit_class TO auth_user;'
+psql postgres://postgres:password@localhost:5432/server -c 'GRANT ALL PRIVILEGES ON TABLE credit_class_version TO auth_user;'
+psql postgres://postgres:password@localhost:5432/server -c 'GRANT ALL PRIVILEGES ON TABLE project TO auth_user;'
+psql postgres://postgres:password@localhost:5432/server -c 'GRANT ALL PRIVILEGES ON TABLE organization TO auth_user;'
```

Comment:
This is a workaround and not the correct privileges. Maybe this is ok for now but not sure why privileges are not correct following the database restore and migrations.

Without this workaround, I am unable to create a project using the local dev environment:


![Screenshot from 2023-08-28 11-51-00](https://github.com/regen-network/regen-web/assets/12519942/eda2411a-3eec-4218-9a1a-6a446b8e13d3)

![Screenshot from 2023-08-28 11-50-20](https://github.com/regen-network/regen-web/assets/12519942/a5fcfa0c-28e7-4744-9c7d-ae577eb21a3c)

Reply:
@wgwz or @blushi any ideas as to what the initialization issue might be here?

Reply:
what's the command you're using to initialize the database from scratch?
see here:
https://github.com/regen-network/regen-server/blob/dev/migrations/README.md#local-development

> 
> Then, you must initialize the database:
> ```
> $ export DATABASE_URL="postgres://postgres:postgres@localhost:5432/regen_registry"
> $ export SHADOW_DATABASE_URL="postgres://postgres:postgres@localhost:5432/regen_registry_shadow"
> $ export ROOT_DATABASE_URL="postgres://postgres:postgres@localhost:5432/postgres"
> $ yarn run graphile-migrate reset --erase
> ```

whereas, any time after that initial setup you need to use `yarn run graphile-migrate migrate` instead (or `yarn run graphile-migrate watch` if you want real-time application of new migrations)

you can see the same lines you added to get it working in the `initial_schema.sql` file:
https://github.com/regen-network/regen-server/blob/dev/migrations/initial_schema.sql#L1175

so it does seem like you might just be running the wrong command?

Reply:
i see the command now, so yeah, if you switch `yarn migrate` to `yarn run graphile-migrate reset --erase` you should be good

Reply:
here's how we set up for CI in the regen-server repo:
https://github.com/regen-network/regen-server/blob/dev/.github/workflows/registryServer.yml#L70

Reply:
Thanks! We do want to run `migrate` at this stage (the first time we start the server container) and avoid resetting the database. This step follows the restoring of the production database (which happens in `db_init.sh`), so this would be similar to us running migrations at the time of promoting a new commit to the production deployment.

I can try to make the setup a bit more clear between init files. Still investigating though.

Reply:
ok, well at some point, in order for the database to be working 100% correct that command does need to get run (otherwise the database would be missing stuff like the privileges). this is just sort of how it works with graphile-migrate. that command only needs to be run once, the first time you are preparing your local database or new database. and then for all subsequent changes after that it's `yarn migrate`.

Reply:
Thanks! This resolved the issue. I had to restructure things a bit to run `pg_restore` within the server container after running `graphile-migrate reset --erase` but that did the trick.